diff --git a/widget/windows/nsLookAndFeel.cpp b/widget/windows/nsLookAndFeel.cpp
index 58c5fc3b55da47109e16cc3d6f44b0950b2965e7..85e3f1a21344678ad0e07f4663fa76653bd7ad3d 100644
--- a/widget/windows/nsLookAndFeel.cpp
+++ b/widget/windows/nsLookAndFeel.cpp
@@ -838,7 +838,7 @@ auto nsLookAndFeel::ComputeTitlebarColors() -> TitlebarColors {
   result.mAccentInactive = dwmKey.GetValueAsDword(u"AccentColorInactive"_ns);
   result.mAccentInactiveText = GetAccentColorText(result.mAccentInactive);
 
-  if (WinUtils::MicaEnabled()) {
+  if (WinUtils::MicaEnabled() || true) {  // Zen: Disable accent color in titlebar because it's ugly
     // Use transparent titlebar backgrounds when using mica.
     result.mActiveDark.mBg = result.mActiveLight.mBg =
         result.mInactiveDark.mBg = result.mInactiveLight.mBg = NS_TRANSPARENT;
