diff --git a/widget/windows/nsLookAndFeel.cpp b/widget/windows/nsLookAndFeel.cpp
index 58c5fc3b55da47109e16cc3d6f44b0950b2965e7..ae1713898d0b8298480bdaf8a9ef7629423c3dee 100644
--- a/widget/windows/nsLookAndFeel.cpp
+++ b/widget/windows/nsLookAndFeel.cpp
@@ -849,6 +849,7 @@ auto nsLookAndFeel::ComputeTitlebarColors() -> TitlebarColors {
   // turned on.
   result.mUseAccent =
       dwmKey.GetValueAsDword(u"ColorPrevalence"_ns).valueOr(0) == 1;
+  result.mUseAccent = false; // Zen: Disable accent color in titlebar because it's ugly
   if (!result.mUseAccent) {
     return result;
   }
