diff --git a/widget/windows/nsWindow.cpp b/widget/windows/nsWindow.cpp
index 00681c216c838efedbbcded9d6b843dbd8b5b0b6..49180ec513155ce262a8928566fc7938ba4a201c 100644
--- a/widget/windows/nsWindow.cpp
+++ b/widget/windows/nsWindow.cpp
@@ -1078,6 +1078,14 @@ nsresult nsWindow::Create(nsIWidget* aParent, const LayoutDeviceIntRect& aRect,
     const DWM_SYSTEMBACKDROP_TYPE tabbedWindow = DWMSBT_TABBEDWINDOW;
     DwmSetWindowAttribute(mWnd, DWMWA_SYSTEMBACKDROP_TYPE, &tabbedWindow,
                           sizeof tabbedWindow);
+
+    // If someone wants to make a patch to firefox, im fine with it. Im tired, boss.
+    // Windows creates some sort of control buttons under the titlebar, which is
+    // annoying for us because it wont allow us to have those hiding animations.
+    // Let's just... remove them... and hope for the best.
+    LONG_PTR style = ::GetWindowLongPtr(mWnd, GWL_STYLE);
+    style &= ~(kTitlebarItemsWindowStyles);
+    ::SetWindowLongPtr(mWnd, GWL_STYLE, style);
   }
 
   if (mOpeningAnimationSuppressed) {
