diff --git a/toolkit/mozapps/extensions/internal/XPIProvider.sys.mjs b/toolkit/mozapps/extensions/internal/XPIProvider.sys.mjs
index 6aeb6a6dc7fd46eefac6a0cd56fadaee11efc6c0..abbd17afaadfadb3f97a0553764ac35aab67116c 100644
--- a/toolkit/mozapps/extensions/internal/XPIProvider.sys.mjs
+++ b/toolkit/mozapps/extensions/internal/XPIProvider.sys.mjs
@@ -2535,8 +2535,8 @@ export var XPIProvider = {
       }
       // Keep version in sync with toolkit/mozapps/extensions/default-theme/manifest.json
       this.maybeInstallBuiltinAddon(
-        "default-theme@mozilla.org",
-        "1.4.1",
+        "firefox-compact-dark@mozilla.org",
+        "1.3",
         "resource://default-theme/"
       );
 
