diff --git a/browser/base/content/browser-box.inc.xhtml b/browser/base/content/browser-box.inc.xhtml
index 108160d9469d44f47c93a3808402e4b27ff59777..7f81b4833bac6b2dac58cb22b0ee71a5a55e8381 100644
--- a/browser/base/content/browser-box.inc.xhtml
+++ b/browser/base/content/browser-box.inc.xhtml
@@ -23,11 +23,15 @@
   </vbox>
   <splitter id="sidebar-splitter" class="chromeclass-extrachrome sidebar-splitter" resizebefore="sibling" resizeafter="none" hidden="true"/>
   <vbox id="appcontent" flex="1">
+    <html:div id="zen-appcontent-navbar-container"></html:div>
     <!-- gNotificationBox will be added here lazily. -->
+    <hbox id="zen-tabbox-wrapper" flex="1">
     <tabbox id="tabbrowser-tabbox"
             flex="1" tabcontainer="tabbrowser-tabs">
+#include zen-tabbrowser-elements.inc.xhtml
       <tabpanels id="tabbrowser-tabpanels"
                   flex="1" selectedIndex="0"/>
     </tabbox>
+  </hbox>
   </vbox>
 </hbox>
