diff --git a/browser/themes/BuiltInThemeConfig.sys.mjs b/browser/themes/BuiltInThemeConfig.sys.mjs
index f738f7f1e37d20d9d9158f052572334714475107..f4b0e9a7077076f2fbe8ac17fa6aadf586c6b37c 100644
--- a/browser/themes/BuiltInThemeConfig.sys.mjs
+++ b/browser/themes/BuiltInThemeConfig.sys.mjs
@@ -33,13 +33,6 @@ export const BuiltInThemeConfig = new Map([
       path: "resource://builtin-themes/dark/",
     },
   ],
-  [
-    "firefox-alpenglow@mozilla.org",
-    {
-      version: "1.4",
-      path: "resource://builtin-themes/alpenglow/",
-    },
-  ],
   [
     "2022red-colorway@mozilla.org",
     {
