diff --git a/mozglue/misc/PreXULSkeletonUI.cpp b/mozglue/misc/PreXULSkeletonUI.cpp
index 416dce0f1f09b2f567ddf4fdab2beab54491a394..f2ffaf839fab724117500a61ce52618c99987277 100644
--- a/mozglue/misc/PreXULSkeletonUI.cpp
+++ b/mozglue/misc/PreXULSkeletonUI.cpp
@@ -862,7 +862,7 @@ Result<Ok, PreXULSkeletonUIError> DrawSkeletonUI(
   ColorRect toolbar = {};
   // In the vertical tabs case the main toolbar is in the titlebar:
   toolbar.color =
-      verticalTabs ? currentTheme.titlebarColor : currentTheme.backgroundColor;
+      true ? currentTheme.titlebarColor : currentTheme.backgroundColor;
   toolbar.x = 0;
   toolbar.y = tabBar.y + tabBarHeight;
   toolbar.width = sWindowWidth;
