diff --git a/browser/components/sidebar/browser-sidebar.js b/browser/components/sidebar/browser-sidebar.js
index 4a124003976684e014435854aef69ce29da541d2..61ce44751c36eea3e5ae2ddcc62e42c01459629b 100644
--- a/browser/components/sidebar/browser-sidebar.js
+++ b/browser/components/sidebar/browser-sidebar.js
@@ -578,7 +578,7 @@ var SidebarController = {
    */
   setPosition() {
     // First reset all ordinals to match DOM ordering.
-    let browser = document.getElementById("browser");
+    let browser = document.getElementById("tabbrowser-tabbox");
     [...browser.children].forEach((node, i) => {
       node.style.order = i + 1;
     });
@@ -592,9 +592,10 @@ var SidebarController = {
       let boxOrdinal = this._box.style.order;
       this._box.style.order = tabbox.style.order;
 
-      tabbox.style.order = boxOrdinal;
       // the launcher should be on the right of the sidebar-box
-      sidebarContainer.style.order = parseInt(this._box.style.order) + 1;
+      this._box.style.order = browser.children.length + 2;
+      this._splitter.style.order = browser.children.length + 1;
+      this._splitter.style.marginRight = 0;
       // Indicate we've switched ordering to the box
       this._box.toggleAttribute("positionend", true);
       sidebarMain.toggleAttribute("positionend", true);
@@ -603,6 +604,9 @@ var SidebarController = {
       this._box.toggleAttribute("positionend", false);
       sidebarMain.toggleAttribute("positionend", false);
       sidebarContainer.toggleAttribute("positionend", false);
+      this._box.style.order = 1;
+      this._splitter.style.order = 2;
+      this._splitter.style.removeProperty("margin-right");
       this.toolbarButton &&
         this.toolbarButton.toggleAttribute("positionend", false);
     }
